src/vcpkg/postbuildlint.cpp

```diff
@@ -1026,73 +1157,87 @@ namespace vcpkg::PostBuildLint
         Util::erase_remove_if(release_libs, lib_filter);
 
         if (!pre_build_info.build_type && !build_info.policies.is_enabled(BuildPolicy::MISMATCHED_NUMBER_OF_BINARIES))
-            error_count += check_matching_debug_and_release_binaries(debug_libs, release_libs);
-
-        if (!build_info.policies.is_enabled(BuildPolicy::SKIP_ARCHITECTURE_CHECK))
         {
-            std::vector<Path> libs;
-            libs.insert(libs.cend(), debug_libs.cbegin(), debug_libs.cend());
-            libs.insert(libs.cend(), release_libs.cbegin(), release_libs.cend());
-            error_count +=
-                check_lib_architecture(pre_build_info.target_architecture, pre_build_info.cmake_system_name, libs, fs);
+            error_count += check_matching_debug_and_release_binaries(debug_libs, release_libs);
         }
 
-        std::vector<Path> debug_dlls = fs.get_regular_files_recursive(debug_bin_dir, IgnoreErrors{});
-        Util::erase_remove_if(debug_dlls, NotExtensionCaseInsensitive{".dll"});
-        std::vector<Path> release_dlls = fs.get_regular_files_recursive(release_bin_dir, IgnoreErrors{});
-        Util::erase_remove_if(release_dlls, NotExtensionCaseInsensitive{".dll"});
-
-        switch (build_info.library_linkage)
+        if (windows_target)
```

Comment:
This if we effectively done on Linux before by the dumpbin value never being filled in.